conversation_reconstruction/dataflow/conversation_reconstruction/dataflow_clean_output.py

```diff
@@ -15,79 +15,77 @@
 
 -------------------------------------------------------------------------------
 
-A dataflow pipeline to collect statistics on Wikipedia talk pages including number of revisions and their sizes.
+A dataflow pipeline to clean scored outputs.
 
 Run with:
 
-  python dataflow_metadata.py --input [YourIngestedFiles] --output [IntendedOutputStorage] --setup_file ./setup.py
-         [Optional] --testmode (runnning locally for test)
+  python dataflow_clean_output.py --setup_file ./setup.py --input=InputStorage --output=OutputStorage --project=YourCloudProject --bucket=YourCloudBucket
 
 """
 from __future__ import absolute_import
 import argparse
 import logging
-import traceback
 import subprocess
-import resource
 import json
 from os import path
 import urllib2
-import ast
-import copy
+import traceback
 import sys
+import multiprocessing
+from construct_utils.utils.third_party.clean import content_clean
 
 import apache_beam as beam
-from apache_beam.io import filesystems
+from apache_beam.metrics.metric import Metrics
 from apache_beam.options.pipeline_options import PipelineOptions
 from apache_beam.options.pipeline_options import SetupOptions
-from construct_utils.conversation_constructor import Conversation_Constructor
-
-
-def get_metadata(x):
-  record_size = len(x)
-  record = json.loads(x)
-  return json.dumps({'page_id': record['page_id'], 'rev_id': record['rev_id'],
-                     'timestamp': record['timestamp'], 'record_size': record_size})
-
 
 def run(known_args, pipeline_args):
-  """Main entry point; defines and runs the reconstruction pipeline."""
+  """Main entry point; defines and runs the pipeline."""
 
   if known_args.testmode:
     pipeline_args.append('--runner=DirectRunner')
   else:
     pipeline_args.append('--runner=DataflowRunner')
+
   pipeline_args.extend([
-    '--project=wikidetox-viz',
-    '--staging_location=gs://wikidetox-viz-dataflow/staging',
-    '--temp_location=gs://wikidetox-viz-dataflow/tmp',
-    '--job_name=get-metadata',
-    '--max_num_workers=80'
-  ])
+    '--project={project}'.format(project=known_args.project),
```

Comment:
Is this really the right way to add arguments? Nearly always is a way to specify default arguments. Maybe use the [python absl argument parser](https://github.com/abseil/abseil-py/blob/master/absl/flags/_flag.py) for example; but I strongly expect the default python argument parser to support default parameters properly too without having to print and then re-parse the parameters.

Reply:
I think the problem here is that there are two groups of arguments mainly, one needs to be sent to dataflow pipeline (the pipeline_args) and the other group is processed by the code (known_args). The main problem here is dividing them into these two groups.